### PR TITLE
[Release] v0.4.2

### PR DESCRIPTION
**Pull request by YunSuk-Yeo**
## Summary of changes

Two hot fixes
* wasm iterator memory leak #440 
* treasury division by zero protection #445 

Improvement
* Bump SDK version to [v0.39.2](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.39.2)
* TaxCaps querier #442 

## Report of required housekeeping

- [x] Github issue OR spec proposal link
- [x] Wrote tests
- [x] Updated API documentation (client/lcd/swagger-ui/swagger.yaml)
- [ ] Added a relevant changelog entry: clog add [section] [stanza] [message]

----

## (FOR ADMIN) Before merging

- [ ] Added appropriate labels to PR
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" (coding standards)
- [ ] Confirm added tests are consistent with the intended behavior of changes
- [ ] Ensure all tests pass

### CODE REVIEW

File: x/treasury/internal/keeper/keeper.go [hunk 1]
```diff
@@ -314,7 +314,7 @@ func (k Keeper) GetTSL(ctx sdk.Context, epoch int64) (res sdk.Int) {
 	bz := store.Get(types.GetTSLKey(epoch))
```
Comment (dokwon) [inline x/treasury/internal/keeper/keeper.go:315]: Fix above comment --> total 'saked' luna --> total staked luna

File: x/treasury/internal/keeper/keeper.go [hunk 2]
```diff
@@ -314,7 +314,7 @@ func (k Keeper) GetTSL(ctx sdk.Context, epoch int64) (res sdk.Int) {
...
 
 	if bz == nil {
-		res = sdk.ZeroInt()
+		res = sdk.OneInt()
```
Comment (dokwon) [inline x/treasury/internal/keeper/keeper.go:317]: Given TSL stands for "Total Staked Luna", it seems to make sense for this GetTSL function to continue to return 0, and have the consumer of this function check for zero value

File: x/treasury/internal/keeper/querier.go [hunk 3]
```diff
@@ -45,6 +47,9 @@ func queryIndicators(ctx sdk.Context, keeper Keeper) ([]byte, error) {
 	taxRewards := sdk.NewDecCoinsFromCoins(keeper.PeekEpochTaxProceeds(ctx)...)
 	TR := keeper.alignCoins(ctx, taxRewards, core.MicroSDRDenom)
 
+	// Subtract remnant blocks of past chain
```
Comment (dokwon) [inline x/treasury/internal/keeper/querier.go:50]: GetCumulatedHeight --> GetCumulativeHeight

File: x/treasury/internal/keeper/querier.go [hunk 4]
```diff
@@ -45,6 +47,9 @@ func queryIndicators(ctx sdk.Context, keeper Keeper) ([]byte, error) {
...
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() - (keeper.GetCumulatedHeight(ctx) % core.BlocksPerWeek))
```
Comment (dokwon) [inline x/treasury/internal/keeper/querier.go:51]: The mod operator and why the ctx construct tolerates a negative value deserves a bit more explanation. Perhaps: 

// The BlockHeight variable of the current context could be set to negative, in which case the querier fetches indicators from epochs corresponding to negative block heights (from a previous chain version)

**Review by dokwon — CHANGES_REQUESTED** [review]
Minor comment and function name changes

**Comment by YunSuk-Yeo** [comment]
All things are updated now @dokwon

File: x/treasury/internal/keeper/indicator.go [hunk 5]
```diff
@@ -68,7 +68,15 @@ func (k Keeper) UpdateIndicators(ctx sdk.Context) {
 
 // TRL returns Tax Rewards per Luna for the epoch
 func TRL(ctx sdk.Context, epoch int64, k Keeper) sdk.Dec {
-	return k.GetTR(ctx, epoch).QuoInt(k.GetTSL(ctx, epoch))
+	tr := k.GetTR(ctx, epoch)
+	tsl := k.GetTSL(ctx, epoch)
+
+	// division by zero protection
+	if tr.IsZero() || tsl.IsZero() {
```
Comment (dokwon) [inline x/treasury/internal/keeper/indicator.go:75]: Perfect

**Review by dokwon — APPROVED** [review]
LGTM

**Pull request merged**